### PR TITLE
Add fake runtimes and CRI changes for KEP-2371

### PR DESCRIPTION
**Pull request by danielye11**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it: 
Add gRPC call and proto for prometheus metrics to CRI API, completing the addition of unstructured metrics described in: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2371-cri-pod-container-stats

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
none
#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```-[KEP]: (https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2371-cri-pod-container-stats) ```

### CODE REVIEW

**Comment by Dragoncell** [comment]
> /test pull-kubernetes-e2e-gce-ubuntu-containerd

Could you please run /retest for this test case ? The failure seems like not related to your change

**Comment by Dragoncell** [comment]
/lgtm
looks good to me overall

**Comment by Dragoncell** [comment]
/lgtm
looks good to me overall

File: staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto [hunk 1]
```diff
@@ -1604,3 +1607,39 @@ enum ContainerEventType {
...
+}
+
+message ContainerMetrics {
+    string container_id = 1;
+    repeated Metric metrics = 2;
+}
+
+message Metric {
+    string name = 1;
+    //return 0 if metric is cached
```
Comment (haircommander) [inline staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto:1630]: nit: can we add a space between // and r

**Comment by haircommander** [comment]
/lgtm

thanks @danielye11

File: staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto [hunk 2]
```diff
@@ -1631,7 +1631,7 @@ message Metric {
     int64 timestamp = 2;
     repeated LabelPair labels = 3;
     MetricType metric_type = 4;
-    Int64Value value = 5;
+    UInt64Value value = 5;
```
Comment (mrunalp) [inline staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto:1634]: Can this be squashed into the first commit?

**Comment by bobbypage** [comment]
These changes were incorporated into https://github.com/kubernetes/kubernetes/pull/113609 which was merged for k8s 1.26.

**Pull request closed without merging**